### PR TITLE
framework/st_things : Adding apis for getting access token and uid of cloud

### PR DESCRIPTION
**Pull request by s-bhuwalka**
Added Api for getting accesss token and uid of cloud So that when called dm_get_access_token()
and dm_get_uid() then it will not return NULL and will get the required values

### CODE REVIEW

**Comment by seinfra** [comment]
Target : [beb10f27f1aa70ab50b73c76c6d61c666df7bf90] - Code Rule Check Result:
framework/src/st_things/things_stack/framework/things_data_manager.c:2469: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/framework/things_data_manager.c:2470: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/src/st_things/things_stack/framework/things_data_manager.c:2480: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/framework/things_data_manager.c:2480: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/framework/things_data_manager.c:2481: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/framework/things_data_manager.c:2481: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/framework/things_data_manager.c:2481: ERROR: [SPC_M_SEP] space prohibited after that open parenthesis '('
framework/src/st_things/things_stack/framework/things_data_manager.c:2482: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/framework/things_data_manager.c:2482: ERROR: [IDT_M_TAB] please, no spaces at the start of a line
framework/src/st_things/things_stack/framework/things_data_manager.c:2482: ERROR: [SPC_M_OPR] spaces required around that '=' (ctx:VxV)
framework/src/st_things/things_stack/framework/things_data_manager.c:2484: ERROR: [IDT_M_TAB] code indent should use tabs where possible
framework/src/st_things/things_stack/framework/things_data_manager.c:2484: ERROR: [IDT_M_TAB] please, no spaces at the start of a line

**Comment by seinfra** [comment]
Target : [beb10f27f1aa70ab50b73c76c6d61c666df7bf90] - Code Rule Check (C++) OK.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 1]
```diff
@@ -2460,3 +2460,27 @@ char *dm_get_model_number()
...
+	}
+	es_cloud_signup_clear(cloud_data);
+	return cloud_access_token;
+}
+//This API will return the uid of the cloud
+//First it will get the cloud credentials from provisioning.dat file and then copies the value of uid in cloud_uid and will return this
+char *dm_get_uid()
+{ 
+	es_cloud_signup_s *cloud_data = NULL;
+        char *cloud_uid;
```
Comment (Hyunjun85) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Please apply tab('\t') instead of spaces(' ').

Comment (Hyunjun85) [reply framework/src/st_things/things_stack/framework/things_data_manager.c]: And It is wrong indentation.

**Comment by seinfra** [comment]
Target : [e8041384cc737b5326040c0f4967d2725658a2c8] - Code Rule Check (C++) OK.

**Comment by SenthilKumarGS** [comment]
@s-bhuwalka 
Prefix the following in the title of the PR.
"framework/st_things:"

Revised title will be: 
framework/st_things: Adding Apis for getting access token and uid of the cloud

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 2]
```diff
@@ -2460,3 +2460,37 @@ char *dm_get_model_number()
 {
 	return g_model_number;
 }
+
+// This API will return the access token of the cloud 
+// First it will get the cloud credentials from provisioning.dat file and
+// then copies the value of access token in cloud_access_token and will return this */
+
```
Comment (SenthilKumarGS) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Remove this empty line.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 3]
```diff
@@ -2460,3 +2460,37 @@ char *dm_get_model_number()
...
+// This API will return the uid of the cloud
+// First it will get the cloud credentials from provisioning.dat file and
+// then copies the value of uid in cloud_uid and will return this
+
+char *dm_get_uid()
+{ 
+	es_cloud_signup_s *cloud_data = NULL;
+	char *cloud_uid;
+	
+	if ((dm_load_legacy_cloud_data(&cloud_data) == 1)) {
```
Comment (SenthilKumarGS) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Two opening paranthesis '((' are not required.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 4]
```diff
@@ -2460,3 +2460,37 @@ char *dm_get_model_number()
...
+// This API will return the access token of the cloud 
+// First it will get the cloud credentials from provisioning.dat file and
+// then copies the value of access token in cloud_access_token and will return this */
+
+char *dm_get_access_token()
+{
+	es_cloud_signup_s *cloud_data = NULL;
+	char *cloud_access_token;
+
+	if ((dm_load_legacy_cloud_data(&cloud_data) == 1)) {
```
Comment (SenthilKumarGS) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Two opening paranthesis '((' are not required.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 5]
```diff
@@ -2460,3 +2460,37 @@ char *dm_get_model_number()
...
+		cloud_access_token = things_strdup(cloud_data->access_token);
+	}
+
+	es_cloud_signup_clear(cloud_data);
+	return cloud_access_token;
+}
+
+// This API will return the uid of the cloud
```
Comment (SenthilKumarGS) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: Move this comment to the header file and make it a multi-line comment.

**Comment by seinfra** [comment]
Target : [e1c4827c3d8e3d266eb0dc56bbd999ed11469bda] - Code Rule Check (C++) OK.

File: framework/src/st_things/things_stack/framework/things_data_manager.h [hunk 6]
```diff
@@ -129,6 +129,19 @@ bool dm_is_es_complete(void);
 bool dm_get_easy_setup_use_artik_crt(void);
 char *dm_get_mnid(void);
 
+/** This API will return the access token of the cloud
```
Comment (abhishek16j) [inline framework/src/st_things/things_stack/framework/things_data_manager.h]: All comments should end with a period (".")/

File: framework/src/st_things/things_stack/framework/things_data_manager.h [hunk 7]
```diff
@@ -129,6 +129,19 @@ bool dm_is_es_complete(void);
...
+ * 
+ * First it will get the cloud credentials from provisioning.dat file and
+ * then copies the value of access token in cloud_access_token and will return cloud_access_token.
```
Comment (abhishek16j) [inline framework/src/st_things/things_stack/framework/things_data_manager.h]: Refactor comments like below: 
1. No need to state the obvious : This API / This function  / This method.
2. No need to reveal platform specific / internal implementation details like file name.


Consider wording like below 

---------------
Returns access token received from cloud.

Internally loads the cloud information stored in persistent storage,
retrieves access token from it and returns the copy of the same. 

*Note: Memory allocated for return parameter should be released by the caller.
-----------------

File: framework/src/st_things/things_stack/framework/things_data_manager.h [hunk 8]
```diff
@@ -129,6 +129,19 @@ bool dm_is_es_complete(void);
...
+ */
+char *dm_get_access_token();
+
+/** This API will return the uid of the cloud
```
Comment (abhishek16j) [inline framework/src/st_things/things_stack/framework/things_data_manager.h]: ditto. please refactor comments as stated for above API.

**Comment by seinfra** [comment]
Target : [ec11ef7b1c1947ad8fe11551da83cd598db90ed4] - Code Rule Check (C++) OK.

**Comment by seinfra** [comment]
Target : [ec11ef7b1c1947ad8fe11551da83cd598db90ed4] - Code Rule Check OK.

File: framework/src/st_things/things_stack/framework/things_data_manager.h [hunk 9]
```diff
@@ -129,6 +129,25 @@ bool dm_is_es_complete(void);
...
+/**
+ * Return uid received from cloud.
+ *
+ * Internally loads the cloud information stored in persistent storage,
+ * retrieves access token from it and returns the copy of the same.
+
+ * Note: Memory allocated for return parameter should be released by the caller.
+ */
+char *dm_get_uid();
 int dm_validate_attribute_in_request(char *res_type, const void *payload);
```
Comment (abhishek16j) [inline framework/src/st_things/things_stack/framework/things_data_manager.h]: add a blank line between two functions for readability.

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 10]
```diff
@@ -2460,3 +2460,29 @@ char *dm_get_model_number()
 {
 	return g_model_number;
 }
+
+char *dm_get_access_token()
+{
+	es_cloud_signup_s *cloud_data = NULL;
+	char *cloud_access_token;
```
Comment (abhishek16j) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: initialize variable to NULL as its used for the return value

File: framework/src/st_things/things_stack/framework/things_data_manager.c [hunk 11]
```diff
@@ -2460,3 +2460,29 @@ char *dm_get_model_number()
...
+	}
+
+	es_cloud_signup_clear(cloud_data);
+	return cloud_access_token;
+}
+
+char *dm_get_uid()
+{ 
+	es_cloud_signup_s *cloud_data = NULL;
+	char *cloud_uid;
```
Comment (abhishek16j) [inline framework/src/st_things/things_stack/framework/things_data_manager.c]: initialize variable to NULL as its used for the return value

**Pull request merged**